utils/misc.go

```diff
@@ -119,3 +120,35 @@ func ConvertTypeToString[T any](value T) string {
 		return fmt.Sprintf("Unsupported type: %v", v.Kind())
 	}
 }
+
+// SortKeys sorts the keys of a map in ascending order.
+func SortKeys[K comparable, V any](mapObj map[K]V) ([]K, error) {
+	if mapObj == nil {
+		return nil, fmt.Errorf("input object cannot be nil")
+	}
+
+	keys := make([]K, 0, len(mapObj))
+	for key := range mapObj {
+		keys = append(keys, key)
+	}
+
+	// Handle empty map gracefully.
+	if len(keys) == 0 {
+		return keys, nil
+	}
+
+	switch reflect.TypeOf(keys[0]).Kind() {
+	case reflect.String:
+		sort.Slice(keys, func(i, j int) bool {
+			return reflect.ValueOf(keys[i]).String() < reflect.ValueOf(keys[j]).String()
+		})
+	case reflect.Uint64:
```

Comment:
Not a huge deal, but why not make this library method work for any type that CanUint() ?

Reply:
I am writing this for the v2 volume support auto salvage feature. We can extend with additional cases in future enhancements.